### PR TITLE
Possibility to specify empty parameters.

### PR DESCRIPTION
**Pull request by chernecov**
1. Parameters is no longer optional (empty string is enough)
2. Validation test has been added.
3. Null returned exception is no longer possible.

### CODE REVIEW

File: Services/StaticJobProvider.php [hunk 1]
```diff
@@ -40,18 +40,22 @@ class StaticJobProvider implements JobProviderInterface
      *
      * @param HostnameManager $hostnameManager
      * @param array           $jobs
+     *
+     * @throws \Exception
```
Comment (oleg-andreyev) [inline Services/StaticJobProvider.php]: exception isn't throw in this method, so this annotation isn't required

File: Services/StaticJobProvider.php [hunk 2]
```diff
@@ -109,4 +113,44 @@ protected function throwRegexMatchException(string $configurationHostname, strin
...
+     *
+     * @param mixed $hostname
+     *
+     * @return void
+     * @throws \Exception
+     */
+    private function validateHostname($hostname)
+    {
+        if (!is_string($hostname)) {
+            throw new \Exception(
```
Comment (oleg-andreyev) [inline Services/StaticJobProvider.php]: imo it's a bad idea to throw generic exceptions

Comment (chernecov) [reply Services/StaticJobProvider.php]: @oleg-andreyev done.

File: Services/StaticJobProvider.php (hunk 1 above)
Comment (chernecov) [reply Services/StaticJobProvider.php]: done.

**Pull request merged**